python/triton/language/core.py

```diff
@@ -1374,65 +1374,88 @@ def maximum(x, y):
 
 
 @jit
-def _max_combine(a, b):
-    return maximum(a, b)
+def _argmax_combine(value1, index1, value2, index2, tie_break_left):
+    if tie_break_left:
+        tie = value1 == value2 and index1 < index2
```

Comment:
This doesn't work for NaN, since NaN != NaN.

Reply:
I will add some docs, saying that the index is still undefined when the max/min value is NaN. It seems more important as it will keep the tie-breaking pretty cheap